code/modules/mob/living/carbon/give.dm

```diff
@@ -1,52 +1,184 @@
+	UnregisterSignal(holder.mob, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS))
+	return ..()
+
+
+/datum/click_intercept/give/InterceptClickOn(mob/user, params, atom/object)
+	if(user == object || !iscarbon(object))
+		return
+	var/mob/living/carbon/receiver = object
+	if(receiver.stat != CONSCIOUS)
+		to_chat(user, "<span class='warning'>[receiver] can't accept any items because they're [receiver == UNCONSCIOUS ? "unconscious" : "dead"]!</span>")
+		return
+	var/obj/item/I = user.get_active_hand()
+	if(!user.Adjacent(receiver))
+		to_chat(user, "<span class='warning'>You need to be closer to [receiver] to offer them [I].</span>")
 		return
+	if(!receiver.client)
+		to_chat(user, "<span class='warning'>You offer [I] to [receiver], but they don't seem to respond...</span>")
+		return
+	// We use UID() here so that the receiver can have more then one give request at one time.
+	// Otherwise, throwing a new "give item" alert would override any current one also named "give item".
+	receiver.throw_alert("give item [I.UID()]", /obj/screen/alert/take_item, alert_args = list(user, receiver, I))
+	item_offered = TRUE // TRUE so we don't give them the default chat message in Destroy.
+	to_chat(user, "<span class='info'>You offer [I] to [receiver].</span>")
+	qdel(src)
+
+
+/**
+ * # Take Item alert
+ *
+ * Alert which appears for a user when another player is attempting to offer them an item.
+ * The user can click the alert to accept, or simply do nothing to not take the item.
+ */
+/obj/screen/alert/take_item
+	name = "Take Item"
+	desc = "someone wants to hand you an item!"
+	icon_state = "template"
+	timeout = 10 SECONDS
+	/// UID of the mob offering the receiver an item.
+	var/giver_UID
+	/// UID of the mob who has this alert.
+	var/receiver_UID
+	/// UID of the item being given.
+	var/item_UID
+	/// If the receiver succesfully took the item or not.
+	var/item_taken = FALSE
+
+
+/obj/screen/alert/take_item/Initialize(mapload, mob/living/giver, mob/living/receiver, obj/item/I)
+	. = ..()
+	desc = "[giver] wants to hand you \a [I]. Click here to accept it!"
+	giver_UID = giver.UID()
+	receiver_UID = receiver.UID()
+	item_UID = I.UID()
+	giver.apply_status_effect(STATUS_EFFECT_OFFERING_ITEM)
+	add_overlay(icon(I.icon, I.icon_state, SOUTH))
+	add_overlay("alert_flash")
+	RegisterSignal(I, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/cancel_give)
+	RegisterSignal(giver, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS), .proc/cancel_give)
+	// If any one of these atoms should be deleted, we need to cancel everything. Also saves having to do null checks before interacting with these atoms.
+	RegisterSignal(I, COMSIG_PARENT_QDELETING, /datum/proc/signal_qdel)
+	RegisterSignal(giver, COMSIG_PARENT_QDELETING, /datum/proc/signal_qdel)
+	RegisterSignal(receiver, COMSIG_PARENT_QDELETING, /datum/proc/signal_qdel)
```

Comment:
These shouldn't be required right? Since you clear the alerts when you qdel a mob now.
Will require some testing though

Reply:
For the receiver, yeah I think I could remove it. Though the giver one is needed because the alert isn't attached to them.